### DISCUSSION TITLE
Question on sharding using vitess on planetscale

### DISCUSSION THREAD

**Discussion by rickythrelfo** [planetscale/discussion#533] [FAQs] [answered]
Is there a way planetscale can shard user data per country and have that data physically located in the cloud region for that user's country to abide by the Data sovereignty laws per country.    Please note, on the surface the dB needs to support queries spanning users in multiple countries.

In planetscale I understand there needs to be a primary region for the database, but i'm hoping that by using sharding and vitess, planetscale  can shard the data into the respective region for that country.  

Is this even possible? Assume the following tables are sharded by country"

CREATE TABLE users (

id CHAR(18) NOT NULL PRIMARY KEY,

user_handle VARCHAR(20),

**country_code VARCHAR(2),**

created_at TIMESTAMP,

PRIMARY KEY (id, country_code)

);

CREATE TABLE blogs (

id CHAR(18) NOT NULL  PRIMARY KEY,

user_id CHAR(18) NOT NULL,

text VARCHAR(280),

country_code VARCHAR(2),

created_at TIMESTAMP

PRIMARY KEY (id, country_code)
);

**Comment by orware** [comment, marked as answer]
Hello @rickythrelfo,

Since we had a more detailed discussion about your questions via your ticket I'll allow that discussion to serve as the answer for this same question you've shared here.

**Reply by rickythrelfo** [reply]
Yes, it's a shame planetscale doesn't support sharding across multi regions (countries) even though vitess does.  I'm hoping one day you do.
